### ISSUE TITLE
[QUESTION] should I align image transform with ultralytics model?

### ISSUE THREAD

**Issue by bluesy7585** [lightly-ai/lightly-train#267]
Hi lightly-train team,

### 🤔 What’s your question?
I'm trying to improve our yolo model with ssl.
I noticed that the default image transform has imagenet mean/std normalization. (train.log)
in my understanding, ultralytics model training does not apply such preprocess.

>         "normalize": {
>             "mean": [
>                 0.485,
>                 0.456,
>                 0.406
>             ],
>             "std": [
>                 0.229,
>                 0.224,
>                 0.225
>             ]
>         }

I wonder should I align the image transform with ultralytics for effective ssl training?
ie. should I remove the normalization?
and another question is does using higher resolution get better performance boost? eg. change 224 -> 320

### 🧠 What did you try?

in my quick experiments, train ssl from COCO pretrain.
then finetune exported model with our data (25k images) using ultralytics training pipeline
```
import lightly_train

if __name__ == "__main__":
    lightly_train.train(
        out="lightly_train_out/exp_11s", 
        data="/home/jovyan/new_repo/ultralytics/humanDet_ssl_ln",
        model="ultralytics/yolo11s.pt",
        method="distillation"     
    )
```
the exported model did not get improvement from ssl. 
the finetuned model has the same performance with simply train from COCO pretrain.

**Comment by yutong-xiang-97** [comment]
Hi @bluesy7585, thank you for reaching out!

For image normalization, Ultralytics also does it but with a different mean and std: 
- https://github.com/ultralytics/ultralytics/blob/025bc04e47bc56f585b179832e5ce09585e60410/ultralytics/data/augment.py#L2643
- https://github.com/orgs/ultralytics/discussions/18977

The reason why we use the ImageNet mean and std is that the teacher is trained with this setting. Using another setting may affect the teacher feature that the student is trained on. During fine-tuning, however, the difference in normalization will be adjusted super quickly. So, essentially, you don't have to do anything.

Also in our internal benchmark, using higher resolution does not give much performance boost. You could ofc explore it on your own to see whether it works for your specific case.

**Comment by liopeer** [comment]
@bluesy7585 I am also interested in what metric you use to evaluate the quality of your model? Is it mAP50-95 or something different? Also, could you have a look at how the other metrics look like, e.g. precision/recall/F1-score and if you there is a difference in false positives/negatives and true positivies/negatives etc. between the COCO checkpoint and the LightlyTrain checkpoint?

**Comment by bluesy7585** [comment]
@yutong-xiang-97 thank you for the clarification.
@liopeer I simply check the mAP50 and mAP50-95 at the end of ultralytics training.
train from COCO pretrain
```
	 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:02<00:00,  5.86it/s]
	   all        500      11077      0.816      0.684      0.772      0.503
```
train from lightly-train exported (100 epochs ssl)
```
	 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:02<00:00,  5.79it/s]
	   all        500      11077      0.819      0.683      0.774        0.5
```
both trained with 60 epochs

**Comment by liopeer** [comment]
Hi @bluesy7585 and thanks for providing the final metrics! If you fine-tune with Ultralytics you should also get a lot of other metrics inside the run directory, like plots of how the losses develop over the training and also F1-curve, Precision-Recall curve and so on. Anything you could provide about that might help narrow the issue down.
And another question: On how many images are you now pretraining, and do you include the finetuning samples as well during pretraining?

**Comment by st22nestrel** [comment]
Hi @yutong-xiang-97 
> Also in our internal benchmark, using higher resolution does not give much performance boost. You could ofc explore it on your own to see whether it works for your specific case.

At my job, we detect logos and run inference using yolo11s with imgsize=1024, as we also want to detect small logos (e.g. small adidas logo that are on football fans t-shirts). I was thinking about feeding images from football games (which is the context where we detect logos) to ligthlytrain for pretraining. It is 125 000 images and I am unable to train with imgsize=1024 on this dataset with 4 H100 GPUs. In docs here https://docs.lightly.ai/train/stable/performance/hardware_recommendations.html you say 4 H100 GPUs should be enough but you do not mention the imgsize you use, probably because you use default transform values, therefore imgsize=224? I think this info could be mention in docs

Do you have any proof that pretraining with imgsize=224 won't have impact on model ability to detect small objects/logos? Most of our images are 1280x720 or 1920x1080 so downscaling this much could sometimes make from visible and distinguishable small logo just a dot.

And do you have any tip on getting the 4 H100 GPUs work with larger imgsize? The memory is the issue here, but I would like to be able to train with same amount of GPUs regardless of the imgsize.

**Comment by liopeer** [comment]
> Do you have any proof that pretraining with imgsize=224 won't have impact on model ability to detect small objects/logos?

In our internal testing we did not see any improvement, which is why we recommend sticking with the 224 resolution, which is also the resolution at which our teacher model performs best. However, we can not rule out that for some data and some models it might have an influence. Since performance (in terms of speed) is also best at 224x224 I would stick with that, but if you're worried about losing too much detail in your images it might be worth trying to change the transform setting for the random resized crop. You can do that via the [`random_resize`](https://docs.lightly.ai/train/stable/train/method_transform_args.html#random-cropping-and-resizing) keyword to the `transform_args` and you can explore the effect it has on the images [here](https://explore.albumentations.ai/transform/RandomResizedCrop).

> And do you have any tip on getting the 4 H100 GPUs work with larger imgsize? The memory is the issue here, but I would like to be able to train with same amount of GPUs regardless of the imgsize.

If you actually want to try with larger images (but I would try the cropping first that I mentioned), you would have to decrease the batch size. But I would definitely not go all the way up to 1280x720 or 1920x1080, but somewhere in between 224 and that.

**Comment by liopeer** [comment]
Maybe also as a quick side note: We crop the image before we do the resizing, so we don't just downsample your images to the resolution of 224x224. This means that at least in some samples, the full detail of the image will be preserved (but it will be a smaller part of the full image). By tweaking the `random_resize` argument you can force the augmentation pipeline to use smaller crops and therefore preserve the details in more of the samples.

**Comment by liopeer** [comment]
Did you have any success with this approach @bluesy7585 ?

**Comment by bluesy7585** [comment]
@liopeer currently I'm not able to get a better performance than COCO pretrained. but I think @yutong-xiang-97 's comment has answered my original question. I will use default image normalization for further experiments. thanks!
